src/components/Text/Base.js

```diff
@@ -52,6 +50,8 @@ const TextBase = ({
       disabled={disabled}
       allCaps={allCaps}
       as={tag}
+      theme={theme}
```

Comment:
I believe it is going to override the theme set with ThemeProvider. I think `themed`  prop alone will be sufficient here. Though for backward compatibility we could create some list of colors/variants to match "traditional". And, I would assume we are moving away from using "traditional" colors, perhaps implement custom prop types validator to warn users that they should be using theme variables.